### PR TITLE
2021 07 todata dg eobj

### PR DESCRIPTION
**Pull request by agenius-logesh**

### CODE REVIEW

File: R/cdfPlot.R [hunk 1]
```diff
@@ -83,15 +87,21 @@ cdfPlot <- function(contrastDF,
...
 
-    assertthat::assert_that(!missing(contrastDF),
-                            !is.null(contrastDF),
-                            "data.frame" %in% class(contrastDF),
-                            nrow(contrastDF) > 0,
-                            msg = "contrastDF must be specified as dataframe with a p-value column.")
+    assertthat::assert_that(!missing(DGEdata),
+                            !is.null(DGEdata),
+                            "DGEobj" %in% class(DGEdata),
+                            msg = "DGEdata must be specified as class of DGEobj.")
```
Comment (saranya-ag) [inline R/cdfPlot.R]: Update to : DGEdata must be specified and must belong to DGEobj class.

File: R/cdfPlot.R [hunk 2]
```diff
@@ -83,15 +87,21 @@ cdfPlot <- function(contrastDF,
...
+
+    assertthat::assert_that(!missing(contrast),
+                            !is.null(contrast),
+                            contrast %in% names(DGEobj::getType(DGEdata, type = "topTable")),
```
Comment (saranya-ag) [inline R/cdfPlot.R]: Check for length of contrast

File: R/cdfPlot.R [hunk 3]
```diff
@@ -83,15 +87,21 @@ cdfPlot <- function(contrastDF,
...
+                            msg = "contrast to be a singular value of class character and must be one from DGEdata with LogIntensity and LogRatio columns and optionally a p-value.")
```
Comment (saranya-ag) [inline R/cdfPlot.R]: Update to : contrast must be a singular value of class character and must be one of the top tables in the DGEdata

File: R/cdfPlot.R [hunk 4]
```diff
@@ -26,7 +26,9 @@
 #' is required for these arguments which applies the attributes in
 #' this order: Significant, Not Significant.
 #'
-#' @param contrastDF A dataframe with LogRatio and LogIntensity columns and optionally a p-value or FDR column.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
+#' @param contrast A character vector of a topTable data in DGEobj and its a class of dataframe
```
Comment (saranya-ag) [inline R/cdfPlot.R]: Name of a topTable item in DGEobj with LogRatio and LogIntensity columns and optionally a p-value or FDR column.

File: R/comparePlot.R [hunk 5]
```diff
@@ -29,8 +29,11 @@
 #' Note: if p-values or FDR values are not used to color the plot, the X Unique color
 #' values are used.
 #'
-#' @param compareDF A dataframe with the first two columns representing the x and y variables.
-#'          Optionally add xp and yp columns to hold p-values or FDR values.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
+#' @param contrastOne and
```
Comment (saranya-ag) [inline R/comparePlot.R]: combine this into one element - Accept a vector of two values.

File: R/comparePlot.R [hunk 6]
```diff
@@ -88,7 +96,10 @@
 #' @importFrom magrittr set_rownames
 #'
 #' @export
-comparePlot <- function(compareDF,
+comparePlot <- function(DGEdata,
+                        contrastOne,
```
Comment (saranya-ag) [inline R/comparePlot.R]: Combine this into one vector.

File: R/plotDispersion.R [hunk 7]
```diff
@@ -69,89 +63,83 @@ plotDispersion <- function(DGEdata,
...
-    assertthat::assert_that(plotCategory %in% c("dispersion", "bcv"),
-                            msg = "Plot Category must  either be dispersion or BCV.")
-
-    if (!is.null(lineFit)) {
-        if (!assertthat::see_if(is.character(linefitColor),
-                                length(linefitColor) == 1)) {
-            warning("linefitColor must be a singular value of class 'character' and must specify the name of the color or the rgb value. Assigning default value 'red'.")
-            linefitColor <- "red"
-        }
+    plotType <- tolower(plotType)
```
Comment (saranya-ag) [inline R/plotDispersion.R:67]: DGEdata is still a DGElist or counts matrix. you have not converted to DGEobj.

File: R/plotNorm.R [hunk 8]
```diff
@@ -38,20 +38,38 @@ plotNorm <- function(DGEdata,
                      plotType = "canvasXpress",
```
Comment (saranya-ag) [inline R/plotNorm.R:38]: plotNorm - Is still accepting a counts matrix or DGEobj. This needs to be accept only DGEobj.

File: tests/testthat/test_cdfPlot.R [hunk 9]
```diff
@@ -2,34 +2,35 @@ context("DGEobj.plots - tests for cdfPlot.R functions")
...
-    skip_if(!("ReplicateGroupDesign_fit" %in% names(t_obj1)))
-
-    top_table <- topTable(t_obj1$ReplicateGroupDesign_fit, number = 100)
+    skip_if(!("BDL_vs_Sham" %in% names(t_obj1)))
 
+    DGEdata = t_obj1
+    contrast = "BDL_vs_Sham"
     # testing plot with default values.
-    plot <- cdfPlot(top_table, referenceLine = "blue")
+    plot <- cdfPlot(DGEdata, contrast, referenceLine = "blue")
```
Comment (saranya-ag) [inline tests/testthat/test_cdfPlot.R]: For all test cases , specify both the parameter name and the value . You are specifying only the value for the first couple of parameters.

File: tests/testthat/test_plotNorm.R [hunk 10]
```diff
@@ -82,10 +82,10 @@ test_that("plotNorm.R: plotNorm()", {
                  regexp = "DGEdata must be of either class 'matrix' or 'DGEobj'.")
     expect_error(plotNorm(NULL),
                  regexp = "DGEdata must be of either class 'matrix' or 'DGEobj'.")
-    expect_error(plotNorm(t_obj1, plotCategory = "heatmap"),
+    expect_warning(plotNorm(t_obj1, plotCategory = "heatmap"),
```
Comment (saranya-ag) [inline tests/testthat/test_plotNorm.R]: check for the output class as well ..

Comment (saranya-ag) [reply tests/testthat/test_plotNorm.R]: Like the other warning statements ..

**Review by saranya-ag — CHANGES_REQUESTED** [review]

File: R/cdfPlot.R (hunk 1 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done for all

File: R/cdfPlot.R (hunk 2 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done for all

File: R/cdfPlot.R (hunk 3 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done for all

File: R/cdfPlot.R (hunk 4 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done

File: R/comparePlot.R (hunk 5 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/comparePlot.R (hunk 6 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/plotDispersion.R (hunk 7 above)
Comment (agenius-logesh) [reply R/plotDispersion.R:67]: done

File: R/plotNorm.R (hunk 8 above)
Comment (agenius-logesh) [reply R/plotNorm.R:38]: done

File: tests/testthat/test_plotNorm.R (hunk 10 above)
Comment (agenius-logesh) [reply tests/testthat/test_plotNorm.R]: done

File: R/cdfPlot.R [hunk 11]
```diff
@@ -26,7 +26,8 @@
 #' is required for these arguments which applies the attributes in
 #' this order: Significant, Not Significant.
 #'
-#' @param contrastDF A dataframe with LogRatio and LogIntensity columns and optionally a p-value or FDR column.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
Comment (agenius-mohammed-ali) [inline R/cdfPlot.R]: I do not think that is an accurate documentation. As user do not pass the object **name** she/he pass the object itself.

File: R/cdfPlot.R [hunk 12]
```diff
@@ -54,16 +55,18 @@
 #'
 #' @examples
 #' \dontrun{
-#'    # Plot to console (contrastDF is a topTable dataframe)
-#'    cdfPlot(contrastDF, title = "My CDF Plot")
+#'    # Plot to console (DGEdata is a name of DGEonj and
+#'    contrast is a dataframe from DGEobj)
+#'    cdfPlot(DGEdata, contrast, title = "My CDF Plot")
```
Comment (agenius-mohammed-ali) [inline R/cdfPlot.R]: - Please clarify in the example what value the **contrast** parameter can hold, so the example can be rerunnable.
-  Add another example for ggplot input

File: R/cdfPlot.R [hunk 13]
```diff
@@ -83,15 +86,22 @@ cdfPlot <- function(contrastDF,
...
+                            length(contrast) == 1,
+                            contrast %in% names(DGEobj::getType(DGEdata, type = "topTable")),
+                            msg = "contrast must be a singular value of class character and must be one of the top tables in the DGEdata.")
+
+    contrastDF <- DGEobj::getItems(DGEdata, contrast)
 
     assertthat::assert_that(!is.null(pvalCol),
                             pvalCol %in% colnames(contrastDF),
-                            msg = "pvalCol column not found in contrastDF.")
+                            msg = "pvalCol column not found in contrast data.")
```
Comment (agenius-mohammed-ali) [inline R/cdfPlot.R]: I think we need to check for the type of contrastDF here as well as the number of rows as old asserts.
(cc. @saranya-ag )

File: R/comparePlot.R [hunk 14]
```diff
@@ -29,8 +29,10 @@
 #' Note: if p-values or FDR values are not used to color the plot, the X Unique color
 #' values are used.
 #'
-#' @param compareDF A dataframe with the first two columns representing the x and y variables.
-#'          Optionally add xp and yp columns to hold p-values or FDR values.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
Comment (agenius-mohammed-ali) [inline R/comparePlot.R]: I do not think that is an accurate documentation. As user do not pass the object name she/he pass the object itself.

File: R/comparePlot.R [hunk 15]
```diff
@@ -58,7 +60,10 @@
 #' @examples
 #' \dontrun{
 #'   # Retrieve the first two contrasts from a DGEobj as a list of dataframes (length = 2; named items)
-#'   contrastList <- getType(DGEobj, "topTable")[1:2]
+#'   contrastList <- lapply(c(contrastOne,contrastTwo), function(x){
```
Comment (agenius-mohammed-ali) [inline R/comparePlot.R]: - Please provide the definition of `contrastOne, contrastTwo` parameters so user can understand how to create them and the example be runnable.
- Please add a ggplot example as well.

File: R/comparePlot.R [hunk 16]
```diff
@@ -67,11 +72,12 @@
 #'   compareDat <- comparePrep(contrastList, significanceCol = "adj.P.Val")
 #'
 #'   # Draw the plot
-#'   cPlot <- comparePlot(compareDat, title = "Plot Title")
+#'   cPlot <- comparePlot(DGEdata, contrasts, title = "Plot Title")
```
Comment (agenius-mohammed-ali) [inline R/comparePlot.R]: - How `contrasts` parameter is initialized, can we clarify that in the example?
- What its relation with the above preparation?

File: R/comparePlot.R [hunk 17]
```diff
@@ -105,14 +113,43 @@ comparePlot <- function(compareDF,
...
+                            !is.null(DGEdata),
+                            "DGEobj" %in% class(DGEdata),
+                            msg = "DGEdata must be specified and must belong to DGEobj class.")
+
+    assertthat::assert_that(!missing(contrasts),
+                            !is.null(contrasts),
+                            length(contrasts) == 2,
+                            contrasts[1] %in% names(DGEobj::getType(DGEdata, type = "topTable")),
+                            contrasts[2] %in% names(DGEobj::getType(DGEdata, type = "topTable")),
+                            msg = "contrast must be a class of character and must be two of the top tables in the DGEdata. with logFC and P.value columns.")
```
Comment (agenius-mohammed-ali) [inline R/comparePlot.R]: I think we need to check that the two columns are of type numeric as well

File: R/comparePlot.R [hunk 18]
```diff
@@ -105,14 +113,43 @@ comparePlot <- function(compareDF,
...
+        compareDF <- comparePrep(contrastList)
+    } else {
+        compareDF <- comparePrep(contrastList)[,1:2]
+    }
+
     plotType <- tolower(plotType)
     assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
                             msg = "Plot type must be either canvasXpress or ggplot.")
+
+
```
Comment (agenius-mohammed-ali) [inline R/comparePlot.R]: extra unneeded space

File: R/comparePlot.R [hunk 19]
```diff
@@ -262,6 +297,7 @@ comparePlot <- function(compareDF,
         dplyr::select(ylabel) %>%
         dplyr::summarise(across(everything(), list(min, max))) %>%
```
Comment (agenius-mohammed-ali) [inline R/comparePlot.R]: - Please add `accross` and `everything` functions to the import above
- Also call `dplyr::` before them

File: R/comparePlot.R [hunk 20]
```diff
@@ -328,7 +364,7 @@ comparePlot <- function(compareDF,
                           symbolColor = symbolColor,
                           symbolFill  = symbolColor)
         # Used to set uniform square scale
-        scalemax = compareDF[,1:2] %>% as.matrix %>% abs %>% max %>% multiply_by(1.05)
+        scalemax = compareDF[,1:2] %>% as.matrix %>% abs %>% max %>% magrittr::multiply_by(1.05)
```
Comment (agenius-mohammed-ali) [inline R/comparePlot.R:269]: - Please add `multiply_by` to the importFrom statements above

File: R/internal.R [hunk 21]
```diff
@@ -40,13 +40,17 @@
...
+    is_valid_shape <- FALSE
+    if (!is.null(shape)) {
+        is_valid_shape <- ifelse(shape %in% valid_shapes, TRUE, FALSE)
+    }
+    is_valid_shape
 }
 
 .get_valid_symbolShapes_cxplot <- function() {
-    valid_shapes <- c("sphere", "square", "rhombus", "triangle", "plus", "star", "octagon", "oval",
+    valid_shapes <- c("circle","sphere", "square", "rhombus", "triangle", "plus", "star", "octagon", "oval",
```
Comment (agenius-mohammed-ali) [inline R/internal.R]: - Defining `valid_shapes` is useless here, you can return the vector directly

File: R/obsPlot.R [hunk 22]
```diff
@@ -1,274 +1,466 @@
...
 #' @param facet Specifies whether to facet (TRUE) or print individual plots
-#'   (FALSE)  (Default = TRUE)
-#' @param facetRow Explicitly set the number of rows for the facet plot.
-#'   Default behavior will automatically set the columns. (Default = NULL)
-#' @param xAngle Angle to set the sample labels on the X axis. (Default =  30; Range = 0-90)
-#' @param scales Specify same scales or independent scales for each subplot (Default = "free_y";
-#'   Allowed values: "fixed", "free_x", "free_y", "free")
+#'   (FALSE)  (default = TRUE)
+#' @param axisFree Specify same scale or independent scales for each subplot (default = TRUE;
+#'   Allowed values: TRUE and FALSE)
```
Comment (agenius-mohammed-ali) [inline R/obsPlot.R]: - Rephrase to: **Allowed values: TRUE or FALSE**

File: R/obsPlot.R [hunk 23]
```diff
@@ -1,274 +1,466 @@
...
+        warning("color specified is not valid. Assigning default value 'deepskyblue3'.")
+        color <- "deepskyblue3"
+    }
+
+    if (violinLayer) {
+        violinColor <- color
+        boxColor <- NA
+    } else {
+        violinColor <- NA
+        boxColor <- color
```
Comment (agenius-mohammed-ali) [inline R/obsPlot.R]: - cannot we use one variable for color instead of two regardless the layer?

File: R/obsPlot.R [hunk 24]
```diff
@@ -1,274 +1,466 @@
...
-                xlab(xlab) +
-                ylab(ylab) +
-                ggtitle(obs) +
-                theme_grey(baseFontSize)
-            aplot <- .addGeoms(aplot)
-
-            # Rotate xaxis group labels
-            if (xAngle > 0) {
-                aplot <- aplot + theme(axis.text.x = element_text(angle = xAngle, hjust = 1))
+        .addGeoms <- function(obsPlot) {
```
Comment (agenius-mohammed-ali) [inline R/obsPlot.R]: - Call `ggplot::` before ggplot related functions

File: R/plotNorm.R [hunk 25]
```diff
@@ -8,7 +8,7 @@
 #' Normalization is performed by edgeR::calcNormFactors. Note TMM is specifically tailored to count-based
 #' data.  Thus this function is only appropriate for count-based data.
 #'
-#' @param DGEdata  A DGEobj or counts matrix.
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
Comment (agenius-mohammed-ali) [inline R/plotNorm.R]: I do not think that is an accurate documentation. As user do not pass the object name she/he pass the object itself.

File: R/plotPValHist.R [hunk 26]
```diff
@@ -4,7 +4,8 @@
 #' numbers. Intended to perform histogram analysis of p-value distributions,
 #' but should be useful for any dataframe of numeric columns.
 #'
-#' @param P.Val A matrix or dataframe of numeric data; col = samples
+#' @param DGEdata Name of DGEobj with a class of DGEobj.
```
Comment (agenius-mohammed-ali) [inline R/plotPValHist.R]: I do not think that is an accurate documentation. As user do not pass the object name she/he pass the object itself.

File: R/plotPValHist.R [hunk 27]
```diff
@@ -16,63 +17,78 @@
...
+    if (any(is.null(P.Val),
+            !is.character(P.Val),
+            length(P.Val) != 1)) {
+        warning("P.Val must be a singular value of class character. Assigning default value 'P.Value'.")
+        P.Val <- "P.Value"
+    }
 
-    assertthat::assert_that(all(sapply(data.frame(P.Val),is.numeric)),
-                            msg = "P.Val must contain only numeric values.")
+    P.Val <- extractCol(getType(DGEdata, "topTable"), colName = P.Val, robust = FALSE)
```
Comment (agenius-mohammed-ali) [inline R/plotPValHist.R]: need to mention what packages function `getType` and `extractCol`  belongs to

File: R/plotPValHist.R [hunk 28]
```diff
@@ -16,63 +17,78 @@
...
     }
 
-    if (!assertthat::see_if(is.logical(facet),
-                            length(facet) == 1)) {
+    if (any(is.null(facet),
+            !is.logical(facet),
+            length(facet) != 1)) {
         warning("facet must be a singular logical value. Assigning default value TRUE.")
-        facet <- TRUE
+        facet = TRUE
```
Comment (agenius-mohammed-ali) [inline R/plotPValHist.R]: Why using "=" sign here instead of the usual `<-`?

**Review by agenius-mohammed-ali — CHANGES_REQUESTED** [review]
Do we need a separate unit tests file for internals.R?
(cc. @saranya-ag )

File: R/comparePlot.R (hunk 18 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/plotPValHist.R (hunk 28 above)
Comment (agenius-logesh) [reply R/plotPValHist.R]: changed.

File: R/plotPValHist.R (hunk 27 above)
Comment (agenius-logesh) [reply R/plotPValHist.R]: done

File: R/cdfPlot.R (hunk 12 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done

File: R/cdfPlot.R (hunk 13 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: I added the check for length.

File: R/cdfPlot.R (hunk 12 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done

File: R/cdfPlot.R (hunk 11 above)
Comment (agenius-logesh) [reply R/cdfPlot.R]: done

File: R/comparePlot.R (hunk 14 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/comparePlot.R (hunk 15 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/comparePlot.R (hunk 16 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/comparePlot.R (hunk 19 above)
Comment (agenius-logesh) [reply R/comparePlot.R]: done

File: R/comparePlot.R (hunk 20 above)
Comment (agenius-logesh) [reply R/comparePlot.R:269]: done

File: R/obsPlot.R (hunk 22 above)
Comment (agenius-logesh) [reply R/obsPlot.R]: done

File: R/obsPlot.R (hunk 23 above)
Comment (agenius-logesh) [reply R/obsPlot.R]: done

File: R/obsPlot.R (hunk 24 above)
Comment (agenius-logesh) [reply R/obsPlot.R]: done

File: R/plotNorm.R (hunk 25 above)
Comment (agenius-logesh) [reply R/plotNorm.R]: done

File: R/plotPValHist.R (hunk 26 above)
Comment (agenius-logesh) [reply R/plotPValHist.R]: done

File: R/comparePlot.R (hunk 17 above)
Comment (agenius-mohammed-ali) [reply R/comparePlot.R]: @agenius-logesh - you missed this one

File: R/internal.R (hunk 21 above)
Comment (agenius-mohammed-ali) [reply R/internal.R]: @agenius-logesh - you missed this one

**Review by agenius-mohammed-ali — COMMENTED** [review]
There are two comments only remaining.

**Pull request merged**